universum/modules/reporter.py

```diff
@@ -149,6 +154,9 @@ def report_build_result(self):
             for observer in self.observers:
                 observer.code_report_to_review(self.code_report_comments)
 
+        if self.settings.fail_unsuccessful and not is_successful:
```

Comment:
Duplication of code is clear indication that something is wrong with this approach